src/util/renderers/renderer.builder.ts

```diff
@@ -0,0 +1,69 @@
+import MediaWiki from './../../MediaWiki.js'
+import { Renderer } from './abstract.renderer.js'
+import { VisualEditorRenderer } from './visual-editor.renderer.js'
+import { WikimediaDesktopRenderer } from './wikimedia-desktop.renderer.js'
+import { RendererBuilderOptions } from './../saveArticles.js'
+import * as logger from './../../Logger.js'
+
+export class RendererBuilder {
+  private renderApi: 'VisualEditor' | 'WikimediaDesktop' | 'WikimediaMobile'
+  private renderMode: 'auto' | 'desktop' | 'mobile' | 'specific'
+
+  public async createRenderer(options: RendererBuilderOptions): Promise<Renderer> {
+    const { RendererMode, RendererAPI } = options
+
+    this.renderMode = RendererMode
+    this.renderApi = RendererAPI
+
```

Comment:
Each function arguments stored 3 times! Why not just define/using the variable directly from the method definition?!